lib/irb.rb

```diff
@@ -613,9 +613,11 @@ def format_prompt(format, ltype, indent, line_no) # :nodoc:
           @context.irb_name
         when "m"
           main_str = @context.safe_method_call_on_main(:to_s) rescue "!#{$!.class}"
+          main_str = main_str || "!#{@context.safe_method_call_on_main(:class)}"
```

Comment:
I think `!` in `"!RuntimeError"` is expressing that stringifying main raised `RuntimeError`, not that main is a `RuntimeError`.
So we don't need `!` prefixed to a non-error class name.

I think main_str can be just an empty string. Just like when `to_s` returned an empty string.
How about `main_str = "#{@context.safe_method_call_on_main(:to_s)}" rescue "!#{$!.class}"`
This way, it can handle other erroneous case such as `to_s` returned `Object.new`

Reply:
Or this one
```ruby
main_str = "!TypeError" unless String === main_str
```

Reply:
Thank you for reviewing.

I misunderstood meaning of prompt.

And `"#{@context.safe_method_call_on_main(:to_s)}"` looks better!
So I changed.